### PR TITLE
feature: make the reader configuration case insensitive for bool

### PR DESCRIPTION
**Pull request by jebrando**

### CODE REVIEW

File: tests/configuration_reader_ut/configuration_reader_ut.c [hunk 1]
```diff
@@ -164,8 +164,8 @@ static void setup_expectation_read_wstring_values(void)
 static void setup_expectation_read_bool_values(void)
 {
     setup_expectation_read_values();
-    STRICT_EXPECTED_CALL(vsprintf_char("%ls", IGNORED_ARG));
-    STRICT_EXPECTED_CALL(free(IGNORED_ARG));
+    //STRICT_EXPECTED_CALL(vsprintf_char("%ls", IGNORED_ARG));
+    //STRICT_EXPECTED_CALL(free(IGNORED_ARG));
```
Comment (dcristoloveanu) [inline tests/configuration_reader_ut/configuration_reader_ut.c:168]: Commented out code detected

File: src/configuration_reader.c [hunk 2]
```diff
@@ -177,44 +177,33 @@ int configuration_reader_get_bool(IFabricCodePackageActivationContext* activatio
...
+            else if (wcscmp(wchar_value, L"") == 0)
+            {
+                *value = false;
+                result = 0;
             }
             else
             {
-                /*Codes_SRS_CONFIGURATION_READER_03_009: [ If the string is False, configuration_reader_get_bool shall set value to false and return 0. ]*/
-                /*Codes_SRS_CONFIGURATION_READER_03_012: [ configuration_reader_get_bool shall succeed and return 0. ]*/
-                if (strcmp(temp, FALSEString) == 0)
```
Comment (dcristoloveanu) [inline src/configuration_reader.c:192]: > if (strcmp(temp, FALSEString) == 0) [](http://example.com/codeflow?start=15&length=36)

Hah, in 2021 and we don't have case insensitive compare in the standard.
Great.

**Review by dcristoloveanu — APPROVED** [review]
:shipit:

File: tests/configuration_reader_ut/configuration_reader_ut.c (hunk 1 above)
Comment (jebrando) [reply tests/configuration_reader_ut/configuration_reader_ut.c:168]: Oops.

**Pull request merged**